cli/glnexus_cli.cc

```diff
@@ -743,25 +759,36 @@ int main_unify_sites(int argc, char *argv[]) {
     }
 
     vector<pair<string,size_t> > contigs;
-    vector<GLnexus::range> ranges;
-    vector<GLnexus::discovered_alleles> valleles;
+    GLnexus::discovered_alleles dsals;
     H("merge discovered allele files",
-      utils::merge_discovered_allele_files(console, discovered_allele_files, contigs, ranges, valleles));
-
-    unsigned discovered_allele_count=0;
-    for (const auto& dsals : valleles) {
-        discovered_allele_count += dsals.size();
+      utils::merge_discovered_allele_files(console, discovered_allele_files, contigs, dsals));
+    console->info() << "consolidated to " << dsals.size() << " alleles for site unification";
+
+    set<GLnexus::range> ranges;
+    if (!bedfilename.empty()) {
+        string range_txt; // empty string, the plan is to get rid of the argument
+        vector<GLnexus::range> i_ranges;
+        H("Parsing range argument", parse_ranges_from_cmd_line(bedfilename, range_txt, contigs, i_ranges));
+        for (auto &r : i_ranges)
+            ranges.insert(r);
     }
-    console->info() << "consolidated to " << discovered_allele_count << " alleles for site unification";
+
+    // find the containing range
+    const GLnexus::range &pos = dsals.begin()->first.pos;
+    GLnexus::range containing_range(-1,-1,-1);
+    H("find containing range", utils::find_containing_range(ranges, pos, containing_range));
 
     vector<GLnexus::unified_site> sites;
-    for (int i = 0; i < valleles.size(); i++) {
-        vector<GLnexus::unified_site> sites_i;
-        H("unify sites", GLnexus::unified_sites(unifier_cfg, valleles[i], sites_i, ranges[i]));
-        sites.insert(sites.end(), sites_i.begin(), sites_i.end());
-    }
+    H("unify sites", GLnexus::unified_sites(unifier_cfg, dsals, sites));
     console->info() << "unified to " << sites.size() << " sites";
 
+    // set the containing ranges for each site
+    for (auto &us : sites) {
+        GLnexus::Status s = utils::find_containing_range(ranges, us.pos, us.containing_target);
```

Comment:
Let's assume that if the user provided any ranges, then all sites should reside within one of those ranges. So instead of the following assert, we can say something like `H("find target range containing site " + us.pos.str(contigs), utils::find_containing_range(ranges, us.pos, us.containing_target));` (if `ranges.size()>0`)